compiler/resolution/functionResolution.cpp

```diff
@@ -8928,10 +8928,9 @@ static void insertUnrefForArrayReturn(FnSymbol* fn) {
           call->insertAtTail(unrefCall);
           FnSymbol* unrefFn = resolveNormalCall(unrefCall);
           resolveFns(unrefFn);
-
-          if (unrefFn->retType == tmp->typeInfo()) {
-            // If the types are equal, we must be dealing with a non-view
-            // array, so we can remove the useless unref call.
+          if (unrefFn->where == NULL) {
```

Comment:
re.  *I ran into odd iterator issues if I removed the chpl__unref for normal arrays* - that is because of the way tuples need chpl__unref - consider a function accepting a tuple argument containing two arrays. The arrays will be passed by reference. If it returns the tuple, the returned tuple needs to store the arrays by value.